datasets/meta_woz/meta_woz.py

```diff
@@ -0,0 +1,153 @@
+quickly simulate user responses with a small amount of data. Such fast-adaptation models fall into the research areas \
+of transfer learning and meta learning. The dataset consists of 37,884 crowdsourced dialogues recorded between two \
+human users in a Wizard of Oz setup, in which one was instructed to behave like a bot, and the other a true human \
+user. The users are assigned a task belonging to a particular domain, for example booking a reservation at a \
+particular restaurant, and work together to complete the task. Our dataset spans 47 domains having 227 tasks total. \
+Dialogues are a minimum of 10 turns long.
+"""
+
+_HOMEPAGE = "https://www.microsoft.com/en-us/research/project/metalwoz/"
+
+_LICENSE = "Microsoft Research Data License Agreement"
+
+_URLs = {
+    "train": "https://download.microsoft.com/download/E/B/8/EB84CB1A-D57D-455F-B905-3ABDE80404E5/metalwoz-v1.zip",
+    "test": "https://download.microsoft.com/download/0/c/4/0c4a8893-cbf9-4a43-a44a-09bab9539234/metalwoz-test-v1.zip",
+}
+
+
+class MetaWoz(datasets.GeneratorBasedBuilder):
+    VERSION = datasets.Version("1.0.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="dialogues", description="The dataset of dialogues from various domains."),
+        datasets.BuilderConfig(
+            name="tasks", description="The metadata for tasks corresponding to dialogues from " "various domains."
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "dialogues"
+
+    def _info(self):
+        if self.config.name == "tasks":
+            features = datasets.Features(
+                {
+                    "task_id": datasets.Value("string"),
+                    "domain": datasets.Value("string"),
+                    "bot_prompt": datasets.Value("string"),
+                    "bot_role": datasets.Value("string"),
+                    "user_prompt": datasets.Value("string"),
+                    "user_role": datasets.Value("string"),
+                }
+            )
+        else:
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "user_id": datasets.Value("string"),
+                    "bot_id": datasets.Value("string"),
+                    "domain": datasets.Value("string"),
+                    "task_id": datasets.Value("string"),
+                    "turns": datasets.Sequence(datasets.Value("string")),
+                }
+            )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download(_URLs)
+        data_dir = dl_manager.extract(data_dir)
```

Comment:
```suggestion
        data_dir = dl_manager.download_and_extract(_URLs)
```
you can do that as well